openedx/core/djangoapps/credit/email_utils.py

```diff
@@ -0,0 +1,138 @@
+from django.core.mail import EmailMessage
+from django.core.urlresolvers import reverse
+from django.utils.translation import ugettext as _
+
+from opaque_keys.edx.keys import CourseKey
+
+from edxmako.shortcuts import render_to_string
+from email.mime.image import MIMEImage
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+from eventtracking import tracker
+from microsite_configuration import microsite
+from openedx.core.djangoapps.user_api.accounts.api import get_account_settings
+from xmodule.modulestore.django import modulestore
+
+
+log = logging.getLogger(__name__)
+
+
+def send_credit_notifications(username, course_key, email_type="credit_eligibility"):
+    """Sends email notification to user on different phases during credit
+    course e.g., credit eligibility, credit payment etc.
+    """
+    try:
+        user = User.objects.get(username=username)
+    except User.DoesNotExist:
+        log.error('No user with %s exist', username)
+        return
+
+    account_settings = get_account_settings(user)
+    course_key = CourseKey.from_string(unicode(course_key))
+    course = modulestore().get_course(course_key, depth=0)
+    course_display_name = course.display_name
+    branded_logo = dict(title='Logo', path=settings.NOTIFICATION_EMAIL_EDX_LOGO, cid=str(uuid.uuid4()))
+    tracking_context = tracker.get_tracker().resolve_context()
+    tracking_id = str(tracking_context.get('user_id'))
+    client_id = str(tracking_context.get('client_id'))
+    events = '&t=event&ec=email&ea=open'
+    tracking_pixel = 'https://www.google-analytics.com/collect?v=1&tid' + tracking_id + '&cid' + client_id + events
+    site_name = microsite.get_value('SITE_NAME', settings.SITE_NAME)
+    dashboard_url_path = reverse('dashboard')
+    courses_url_path = reverse('courses') + "?type=credit"
+    dashboard_link_parts = ("https", site_name, dashboard_url_path, '', '', '')
+    dashboard_link = urlparse.urlunparse(dashboard_link_parts)
+    credit_course_link_parts = ("https", site_name, courses_url_path, '', '', '')
+    credit_course_link = urlparse.urlunparse(credit_course_link_parts)
+    context = {
+        'full_name': account_settings['name'],
+        'platform_name': settings.PLATFORM_NAME,
+        'course_name': course_display_name,
+        'branded_logo': branded_logo['cid'],
+        'dashboard_link': dashboard_link,
+        'credit_course_link': credit_course_link,
+        'tracking_pixel': tracking_pixel,
+    }
+
+    # create the root email message
+    notification_msg = MIMEMultipart('related')
+    # add 'alternative' part to root email message to encapsulate the plain and
+    # HTML versions, so message agents can decide which they want to display.
+    msg_alternative = MIMEMultipart('alternative')
+    notification_msg.attach(msg_alternative)
+    # render the credit notification templates
+    subject = None
+    if email_type == "credit_eligibility":
```

Comment:
I'm not sure that this is useful now that we're sending just the credit eligibility email here.  You could maybe abstract this a little and pass in the two template paths and subject instead?